lib/pdk/util/git.rb

```diff
@@ -19,6 +19,8 @@ def initialze(args, result)
     end
 
     module Git
+      GIT_QUERY_CACHE_TTL ||= 10
```

Comment:
The use case I was worried about was:

1. Check if a directory, which doesn't exist, is a git repo  (e.g. /foo)
2. Then git clone into that directory (git clone ...uri...  /foo)

Note that `.clone` isn't a method in the Git module.

foo is _NOW_ a valid git repo.

Yeah I'm not 100% happy with using an "arbitrary" time out either, but I couldn't figure out any other way to invalidate the cache.  There are _significant_ improvements with caching though.

10s was, in my mind, long enough to provide a performance benefit but slow enough so that it would be meaningful.

Open to other ideas on how to do this 👍

Reply:
I see.

I would say that down the line it might be good to route all git operations through this module and then the methods that impact the validity of the cache can just handle the invalidation internally as part of like `PDK::Util::Git.clone` or whatever. Perhaps we should just add a maintenance ticket to centralize all Git operations like we did for filesystem stuff?

Reply:
Raised https://tickets.puppetlabs.com/browse/PDK-1551 to track the work